### DISCUSSION TITLE
Messy server, where to begin?

### DISCUSSION THREAD

**Discussion by adinunzio10** [immich-app/immich#3811] [Q&A] [answered]
Long story short, I wiped out all my docker volumes under the impression I had them backed up via borg. I didn't. (Now fixed to properly backup) 

Most of my dockers were OK as the stuff in volumes wasn't anything critical from the looks of it. Except Immich.

When I logged into Immich I had no pics, but still using ~90GB of data.  Stupidly I went into the immich-server and ran an API request to upload and import all my pictures from the upload_directory.  Now when I go into immich I see duplicates of everything and the interface is mega-slow.  It also now says I have ~130GB of data, so not  everything must have duplicated.  I noticed a bunch of files in my upload_directory with a +1 or edited+1. I removed all of them. I also saw a ton of webp duplicates, removed all of them.   

Server stats page says I have 88GB (which is the number I had before I caused this giant mess of work for myself) , but in the bottom left corner it says 132GB.

du -sh of the upload_directory  also says 132GB. 

The .webp files stil show on my immich home screen, despite their removal. I set it to re-gen all thumbnails so not sure if that will work. It says it still has 20k left so I will have to come back and check later.  

Any help to point me in the right direction to bring immich back to its greatness for me. I know it's user error so just need to clean the mess. Live and learn.

**Comment by jrasm91** [comment]
I would probably would have recommend just reimporting everything like you did, but using the --import option with --no-read-only. At this point, it is a bit tricky since the state of the filesystem is in question. It might be easier now to use a new folder for your UPLOAD_LOCATION and reupload everything from your old library. Doing it this way will help you avoid having duplicates in the new library.

**Reply by adinunzio10** [reply]
Yeah I was reading through doco while the import was going and even posted a few hours ago on mastodon that I bet I was supposed to use the --import option.  once again live and learn haha. 

Okay I am going to try setting up a new upload location and reupload and will mark as answer once confirmed I am good to go! 

Thanks!

**Reply by jrasm91** [reply]
It probably goes without saying, but I would also recommend reading through this section 😁

https://immich.app/docs/administration/backup-and-restore

The database is actually really easy to backup and using pgdump is generally more reliable and robust than a volume backup.

**Reply by adinunzio10** [reply]
I did read through that one as well (after all the mess haha) and will do the pgdump ontop of my borg backups to be safe. 

and in other news, I forgot that while I may have deleted my docker volumes, I still have backups of the upload_location directory. Looking at the one from yesterday and its the 94GB I expected. So I restored that backup and then reran docker-compose up and everything still seemed a mess on the web ui. 

So I wiped out my immich volumes and rebuilt them and then ran the import again with --import (--no-read-only didn't work for me or i did it wrong) 
it identifies all of my assets in the restored upload location. imports them, and once completed, then screen flooded with errors saying File does not exist within user's external path and then no files appear in the web UI

**Reply by adinunzio10** [reply]
Okay I think I am a little confused on how things work and need to do some more reading again.  I created a new folder to dedicate as my new immich upload_location. Then added my old upload_location (that I restored from yesterdays backup) as a mount for the immich-server and immich-microservices. Then I ran an upload without the --import flag. 

I think I was confused to the point where no, I do not have a backup of the actual file structure immich needs (as that would have been stored in the database volume pgdata, which I lost) so for immich to reupload my files into the proper file structure I restored from yesterday it needs to go through a full upload again.   I _believe_ I should be good to go at the end of this? Hopefully.

**Reply by jrasm91** [reply]
Yes that sounds right. Immich needs the database to be restored or you need to reimport everything again.

**Reply by adinunzio10** [reply]
Something still doesn't seem right.  I compared the file structure of my old folder to the new one and they are identical.  Either way it brought in some new issues.

I still seem to have duplicates, a webp and a jpg of the same image. I think the webp is for thumbnals? but both are showing as seperate files. 
Interface still seems slower then it was before the big mess i caused.
Interface shows a portion of dead thumnbails 
A lot of my pics dates reset to yesterdays date. 

Ugh this doesn't make sense, I started over from scratch, i really thought id be okay now.  I am sure it's user error, I might just leave it as it is for now.. 

It looks like even the photos that were used as thumbnails for people's faces are showing as their own image. 

Update: If i scroll passed yesterdays date in my viewer, it looks a lot better, no more duplicates and proper date stamps.  Maybe I'll just delete all pics from yesterdays date from immich. If anything legit was in there I still have backups of my photos from before this mess happened.

**Reply by bo0tzz** [reply, marked as answer]
Sounds like you reuploaded the whole UPLOAD_LOCATION backup, including all the thumbnail folders etc. You should only upload the `library` folder, that's where your original files live.

**Reply by adinunzio10** [reply]
That makes sense. I will copy over my backup of just the library folder and try again

**Reply by adinunzio10** [reply]
Okay hopefully this is the end of my messy adventure.  I removed all immich volumes. Cleared out the upload_location completely, and added my old library (only the library folder) as a mount in the docker as /usr/src/temp/

Booted up the docker, logged into immich to create my account and get API key
started a shell of the immich_server docker and ran 'immich upload --key mynewkeyhere --server http://10.0.0.96:2283/api --recursive /usr/src/temp/ 

It's importing 8k assets, which is the number I was expecting (makes sense why I was getting so many more before since I was treating the thumbnails and what not as their own library like you said.

once its done, il remove the mounted path from the compose file and should be good. to. go...

**Reply by adinunzio10** [reply]
Thank you both @bo0tzz  and @jrasm91 for sticking with me through my countless mistakes. Everything looks back to normal again! yay! and I learned a bit more how immich file strucutre and backups work. 

To recap if I run into it again or someone else: 

After losing all my dockers. I should have just spun up a new immich container with a new immich upload_directory and set the `library` folder of my old directory as a mounted path in the server and microservices containers.  then i went into the server docker and ran an immich api upload from the bulk CLI documentation.
